src/CMakeLists.txt

```diff
@@ -118,6 +120,7 @@ target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
     ignition-math${IGN_MATH_VER}::ignition-math${IGN_MATH_VER}
     ignition-utils${IGN_UTILS_VER}::ignition-utils${IGN_UTILS_VER}
   PRIVATE
+    ignition-common${IGN_COMMON_VER}::ignition-common${IGN_COMMON_VER}
```

Comment:
we added `ignition::common` for USD I'm not really sure that you can use this dependency here

Reply:
The dependency doesn't seem bad to me. Here is the code I'd duplicate, which I could trim (haha) down:

```
/////////////////////////////////////////////////
void ignition::common::ltrim(std::string &_s)
{
  _s.erase(_s.begin(), std::find_if(_s.begin(), _s.end(),
        [](int c) {return !std::isspace(c);}));
}

/////////////////////////////////////////////////
void ignition::common::rtrim(std::string &_s)
{
  _s.erase(std::find_if(_s.rbegin(), _s.rend(),
        [](int c) {return !std::isspace(c);}).base(),
      _s.end());
}

/////////////////////////////////////////////////
void ignition::common::trim(std::string &_s)
{
  ignition::common::ltrim(_s);
  ignition::common::rtrim(_s);
}

/////////////////////////////////////////////////
std::string ignition::common::ltrimmed(std::string _s)
{
  ignition::common::ltrim(_s);
  return _s;
}   

/////////////////////////////////////////////////
std::string ignition::common::rtrimmed(std::string _s)
{
  ignition::common::rtrim(_s);
  return _s;
}

/////////////////////////////////////////////////
std::string ignition::common::trimmed(std::string _s)
{
  ignition::common::trim(_s);
  return _s;
}
```

Utilizing existing code rather than duplicating it seems like something we should do internally. Otherwise, we should be telling users to copy our code, and not use the libraries.

Reply:
I don't think we should add a dependency in a stable version. We already have `sdf::trim` that can be used instead.

Reply:
Good point, this is a stable version.

Reply:
Done in [e6fc820](https://github.com/ignitionrobotics/sdformat/pull/911/commits/e6fc820d074fc8f07a654210142e147a69c909d9)

Reply:
Adding `ign-common` as a dependency to the core of SDF is something we should think carefully about. That brings a lot of dependencies. An alternative to consider would be to move some of this functionality to `ign-utils`.